sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2DataFrameSuite.scala

```diff
@@ -263,4 +264,38 @@ class DataSourceV2DataFrameSuite
       spark.listenerManager.unregister(listener)
     }
   }
+
+  test("columns with default values") {
+    val tableName = "testcat.ns1.ns2.tbl"
+    withTable(tableName) {
+      sql(s"CREATE TABLE $tableName (id INT, dep STRING) USING foo")
```

Comment:
nit: let's also test creating a V2 table with column default values, and then inserting into the V2 table

Reply:
Sure, added a test below to verify CREATE/REPLACE statements.